src/Data/Nat/Base.agda

```diff
@@ -130,6 +132,47 @@ instance
 >-nonZero⁻¹ : ∀ n → .{{NonZero n}} → n > 0
 >-nonZero⁻¹ (suc n) = z<s
 
+-- The property of being a non-zero, non-unit
+
+pattern 2+ n = suc (suc n)
+
+trivial : ℕ → Bool
+trivial 0      = true
+trivial 1      = true
+trivial (2+ _) = false
+
+NonTrivial : Pred ℕ 0ℓ
+NonTrivial = T ∘ not ∘ trivial
+
+instance
+  nonTrivial : ∀ {n} → NonTrivial (2+ n)
+  nonTrivial = _
+
+pattern 1<2+n {n} = s<s (z<s {n})
+
+-- Constructors
+
+n>1⇒nonTrivial : ∀ {n} → 1 < n → NonTrivial n
+n>1⇒nonTrivial 1<2+n = _
+
+nonZero⇒≢1⇒nonTrivial : ∀ {n} → .⦃ NonZero n ⦄ → n ≢ 1 → NonTrivial n
+nonZero⇒≢1⇒nonTrivial {1}    = contradiction refl
+nonZero⇒≢1⇒nonTrivial {2+ _} = _
+
+recompute-nonTrivial : ∀ {n} → .⦃ NonTrivial n ⦄ → NonTrivial n
+recompute-nonTrivial {2+ _} = _
+
+-- Destructors
+
+nonTrivial⇒nonZero : ∀ {n} → .⦃ NonTrivial n ⦄ → NonZero n
+nonTrivial⇒nonZero {n = 2+ k} = _
+
+nonTrivial⇒n>1 : ∀ n → .⦃ NonTrivial n ⦄ → 1 < n
+nonTrivial⇒n>1 (2+ _) = 1<2+n
+
+nonTrivial⇒≢1 : ∀ {n} → .⦃ NonTrivial n ⦄ → n ≢ 1
```

Comment:
This `n` should be explicit!

Reply:
Surely not? `n ≢ 1` is a negated proposition, so any assumption of the form `n ≡ 1` towards a contradiction will have instantiated `n`? See also eg. `Data.Nat.Base.≢-nonZero⁻¹` which never, AFAICT, needs its argument explicit except in the definition...?